server/services/database/documents.js

```diff
@@ -19,27 +18,33 @@ const getAllDocuments = () => {
       inner join users as u\
       on u.id = d.owner_id;";
 
-    pool.query(sqlQuery, (error, result) => {
-      if (error) {
-        reject(error);
-      } else {
-        resolve(result.rows);
-      }
-    });
-  });
+  return pool
+    .query(sqlQuery)
+    .then(result => result.rows)
+    .catch(e => console.error(e));
 };
 
 const getUserDocuments = userId => {
-  const documents = "SELECT * FROM documents WHERE owner_id=$1";
-  return new Promise((resolve, reject) => {
-    pool.query(documents, [userId], (error, result) => {
-      if (error) {
-        reject(error);
-      }
+  const sqlQuery =
+    "select\
+          d.id, d.format, d.name,\
+          d.status, d.submission_date, d.due_date,\
+          lf.name as from_language_name,\
+          lt.name as to_language_name,\
+          u.name as owner_name\
+      from documents as d\
+      inner join languages as lf\
+      on  lf.code = d.from_language_code\
+      inner join languages as lt\
+      on lt.code = d.to_language_code\
+      inner join users as u\
+      on u.id = d.owner_id\
+      where d.owner_id=$1;";
```

Comment:
This SQL query is exactly the same as the one used for interpreters, except you add a where condition at the end. You could extract the common part of the query outside these functions to share it and avoid code duplication. And for getUserDocuments, you just need to append the where condition.

Reply:
Should I extract the common part of the query just outside functions or in different file?

Reply:
@kosenchiha outside the function is fine for now :) If the documents.js becomes too big at some point, we can think about extracting things in different files. but for now it's okay.